persistence.js

```diff
@@ -536,10 +537,54 @@ class RedisPersistence extends CachedPersistence {
     pipeline.sadd(SHAREDTOPICS, topic)
     pipeline.sadd(topic, groupTopic)
     pipeline.sadd(groupTopic, clientTopic)
+    // Adding each shared topic to the list where it will be wiped if it will be not updated within 20 secs.
+    // Broker from his side should update all shared topics each 10 seconds.
+    // So if broker and client alive - topic will persist, if not - will be wiped
+    // Protection from leackage of shared subscriptions, in case if client is already dead but topic somehow
+    // left on the list
+    pipeline.zadd(SHAREDTOWIPE, (new Date() / 1000) + 20, `${groupTopic}@${clientTopic}`)
     pipeline.exec((err) => {
       if (err) {
         return cb(err)
       }
+
+      if (!this._clenupOldShared) {
+        this._clenupOldShared = setInterval(() => {
+          const luaScript = `
+            local sharedtowipe = KEYS[1]
+            local sharedTopics = KEYS[2]
+            local currentTime = tonumber(ARGV[1])
+            local deleted = {}
+            local elements = redis.call("zrangebyscore", sharedtowipe, "-inf", currentTime)
+            for _, element in ipairs(elements) do
+              local parts = {}
+              for str in string.gmatch(element, "([^@]+)") do
+                table.insert(parts, str)
+              end
+              local groupTopic, clientTopic = parts[1], parts[2]
+              local parts2 = {}
+              for str in string.gmatch(groupTopic, "([^_]+)") do
+                table.insert(parts2, str)
+              end
+              local originalTopic = parts2[2]
+              redis.call("srem", groupTopic, clientTopic)
+              redis.call("zrem", sharedtowipe, element)
+              table.insert(deleted, clientTopic)
+              local groupCardinality = redis.call("scard", groupTopic)
+              if groupCardinality == 0 then
+                redis.call("srem", originalTopic, groupTopic)
+                local topicCardinality = redis.call("scard", originalTopic)
+                if topicCardinality == 0 then
+                  redis.call("srem", sharedTopics, originalTopic)
+                end
+              end
+            end
+
+            return deleted
+          `
+          this._db.eval(luaScript, 2, [SHAREDTOWIPE, SHAREDTOPICS, (new Date() / 1000)], () => {})
+        }, 1) // Each second check should we remove some outdated shared subscription or not
```

Comment:
The delay is in milliseconds (https://nodejs.org/api/timers.html#setintervalcallback-delay-args)
```suggestion
        }, 1000) // Each second check should we remove some outdated shared subscription or not
```